opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/hystrix/SleuthHystrixConcurrencyStrategy.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2013-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentracing.contrib.spring.cloud.hystrix;
+
+import com.netflix.hystrix.HystrixThreadPoolKey;
+import com.netflix.hystrix.strategy.HystrixPlugins;
+import com.netflix.hystrix.strategy.concurrency.HystrixConcurrencyStrategy;
+import com.netflix.hystrix.strategy.concurrency.HystrixRequestVariable;
+import com.netflix.hystrix.strategy.concurrency.HystrixRequestVariableLifecycle;
+import com.netflix.hystrix.strategy.eventnotifier.HystrixEventNotifier;
+import com.netflix.hystrix.strategy.executionhook.HystrixCommandExecutionHook;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisher;
+import com.netflix.hystrix.strategy.properties.HystrixPropertiesStrategy;
+import com.netflix.hystrix.strategy.properties.HystrixProperty;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.concurrent.TracedCallable;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link HystrixConcurrencyStrategy} that wraps a {@link Callable} in a
+ * {@link Callable} that either starts a new span or continues one if the tracing was
+ * already running before the command was executed.
+ *
+ * @author Marcin Grzejszczak
+ * NOTE: Modified original by @author kameshsampath to suit OpenTracing
+ */
+public class SleuthHystrixConcurrencyStrategy extends HystrixConcurrencyStrategy {
+
+	private static final String HYSTRIX_COMPONENT = "hystrix";
+	private static final Log log = LogFactory
+			.getLog(SleuthHystrixConcurrencyStrategy.class);
+
+	private final Tracer tracer;
+	private HystrixConcurrencyStrategy delegate;
+
+	public SleuthHystrixConcurrencyStrategy(Tracer tracer) {
```

Comment:
This is very similar to https://github.com/OpenFeign/feign-opentracing/blob/master/feign-hystrix-opentracing/src/main/java/feign/opentracing/hystrix/TracingConcurrencyStrategy.java which is registered here https://github.com/opentracing-contrib/java-spring-cloud/blob/master/opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/feign/FeignTracingAutoConfiguration.java#L46

That said, we have to make sure it works well together

Reply:
IMHO - lets have it separate so that changes to each specific functionality does not need regression on other ..